bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -60,6 +61,11 @@ public class FindReplaceLogic implements IFindReplaceLogic {
 
 	@Override
 	public void setFindString(String findString) {
+		if (this.findString.isEmpty() && !findString.isEmpty()) {
+			// User just started a new search after clearing previous search.
+			if (target != null)
+				restoreBaseLocation = target.getSelection();
```

Comment:
Please always put expression blocks into braces (applies in the same way also to other places in the PR).
```suggestion
			if (target != null) {
				restoreBaseLocation = target.getSelection();
			}
```

Reply:
Taken care, thanks.